subiquity/server/tests/test_server.py

```diff
@@ -40,55 +42,73 @@ def setUp(self):
-        autoinstall = self.create(self.path('arg.autoinstall.yaml'))
+        self.create(reload_autoinstall_path, 'reload')
+        autoinstall = self.create(self.path('arg.autoinstall.yaml'), 'arg')
         self.server.opts.autoinstall = autoinstall
-        self.create(cloud_autoinstall_path)
-        self.create(iso_autoinstall_path)
-        self.assertEqual(expected, self.server.select_autoinstall_location())
+        self.create(cloud_autoinstall_path, 'cloud')
+        iso = self.create(iso_autoinstall_path, 'iso')
+        self.assertEqual(iso, self.server.select_autoinstall())
+        self.assert_contents(iso, 'reload')
 
     def test_arg_wins(self):
-        expected = self.create(self.path('arg.autoinstall.yaml'))
-        self.server.opts.autoinstall = expected
-        self.create(cloud_autoinstall_path)
-        self.create(iso_autoinstall_path)
-        self.assertEqual(expected, self.server.select_autoinstall_location())
+        arg = self.create(self.path('arg.autoinstall.yaml'), 'arg')
+        self.server.opts.autoinstall = arg
+        self.create(cloud_autoinstall_path, 'cloud')
+        iso = self.create(iso_autoinstall_path, 'iso')
+        self.assertEqual(iso, self.server.select_autoinstall())
+        self.assert_contents(iso, 'arg')
 
     def test_cloud_wins(self):
-        expected = self.create(cloud_autoinstall_path)
-        self.create(iso_autoinstall_path)
-        self.assertEqual(expected, self.server.select_autoinstall_location())
+        self.create(cloud_autoinstall_path, 'cloud')
+        iso = self.create(iso_autoinstall_path, 'iso')
+        self.assertEqual(iso, self.server.select_autoinstall())
+        self.assert_contents(iso, 'cloud')
 
     def test_iso_wins(self):
-        expected = self.create(iso_autoinstall_path)
-        self.assertEqual(expected, self.server.select_autoinstall_location())
+        iso = self.create(iso_autoinstall_path, 'iso')
+        self.assertEqual(iso, self.server.select_autoinstall())
+        self.assert_contents(iso, 'iso')
 
     def test_nobody_wins(self):
-        self.assertIsNone(self.server.select_autoinstall_location())
-
-    def test_copied_to_reload(self):
-        self.server.autoinstall = self.tmp_path('test.yaml', dir=self.tempdir)
-        expected = 'stuff things'
-        with open(self.server.autoinstall, 'w') as fp:
-            fp.write(expected)
-        self.server.save_autoinstall_for_reload()
-        with open(self.path(reload_autoinstall_path), 'r') as fp:
-            self.assertEqual(expected, fp.read())
+        self.assertIsNone(self.server.select_autoinstall())
 
     def test_bogus_autoinstall_argument(self):
         self.server.opts.autoinstall = self.path('nonexistant.yaml')
         with self.assertRaises(Exception):
-            self.server.select_autoinstall_location()
+            self.server.select_autoinstall()
+
+    def test_early_commands_changes_autoinstall(self):
```

Comment:
Could you add a test or modify this test to simulate the autoinstall config being supplied via cloud-init config?

Reply:
Good point, it's the entire reason for the PR.  Done!